Npm Extractor - allow JSON output only if it was required

liron-shalom opened the pull request.

yahavi commented on build-info-extractor-npm/src/main/java/org/jfrog/build/extractor/npm/extractor/NpmBuildInfoExtractor.java:
```diff
@@ -159,6 +162,10 @@ private void createTempNpmrc(Path workingDir, List<String> installationArgs) thr
     }
 
+    private boolean isJsonOutputRequiered(List <String> installationArgs) {
+        return installationArgs.contains("--json") || installationArgs.contains("--json=true");
```
1. `installationArgs.contains("--json=true")` infer `installationArgs.contains("--json")`. `installationArgs.contains("--json")` is enough.
2. This will not work on 
```
--json=false
--json false
--json        false
--json true
--json        true
```

yahavi commented on build-info-extractor-npm/src/main/java/org/jfrog/build/extractor/npm/extractor/NpmBuildInfoExtractor.java:
```diff
@@ -159,6 +162,20 @@ private void createTempNpmrc(Path workingDir, List<String> installationArgs) thr
+        if (jsonIndex > -1 ) {
+            return jsonIndex == installationArgs.size() - 1 || !installationArgs.get(jsonIndex).equals("false");
+        }
+        return installationArgs.contains("--json=true");
```
If `jsonIndex == -1`, we don't have `--json` in `installationArgs`.
Conclusion: `installationArgs.contains("--json=true")` is always false.

yahavi commented on build-info-extractor-npm/src/main/java/org/jfrog/build/extractor/npm/extractor/NpmBuildInfoExtractor.java:
```diff
@@ -159,6 +162,20 @@ private void createTempNpmrc(Path workingDir, List<String> installationArgs) thr
+    private boolean isJsonOutputRequired(List <String> installationArgs) {
+        int jsonIndex = installationArgs.indexOf("--json");
+        if (jsonIndex > -1 ) {
+            return jsonIndex == installationArgs.size() - 1 || !installationArgs.get(jsonIndex).equals("false");
```
I wonder if this code would return false if we have`--json=false`. 🤔 
I think that this function deserved a unit test.

liron-shalom commented:
We split the raw string args into a list by whitespace:
` this.installArgs = StringUtils.isBlank(installArgs) ? new ArrayList<>() : Arrays.asList(installArgs.trim().split("\\s+"));`
so the case of "--arg=value" is treated as one element in the list/
